### DISCUSSION TITLE
SeaORM Relations Help

### DISCUSSION THREAD

**Discussion by HTGAzureX1212** [SeaQL/sea-orm#1709] [Q&A]
I currently have 5 tables that have certain relations with each other:

1. `workflow` table
2. `build` table, which has a foreign key to the `workflow` table
3. `approve_build` table, which has a foreign key to the `workflow` table
4. `pull_request` table, which has foreign keys to `approve_build` and `build` tables
5. `enqueued` table, which has a foreign key to the `pull_request` table

My question is, how can I express a query that selects everything from the enqueued table, while also resolving all the foreign key models, using join? This is quite a lot of layers of foreign keys and I wonder if SeaORM supports that with an ORM API or do I need to make use of lower level APIs or even raw queries to get this done.

**Comment by billy1624** [comment]
Hey @HTG-YT, try `DataLoader` https://www.sea-ql.org/SeaORM/docs/relation/data-loader/
